app/src/main/java/com/tari/android/wallet/infrastructure/backup/WalletBackup.kt

```diff
@@ -0,0 +1,77 @@
+package com.tari.android.wallet.infrastructure.backup
+
+import com.tari.android.wallet.infrastructure.backup.compress.CompressionMethod
+import com.tari.android.wallet.infrastructure.security.encryption.SymmetricEncryptionAlgorithm
+import net.lingala.zip4j.ZipFile
+import net.lingala.zip4j.model.ZipParameters
+import net.lingala.zip4j.model.enums.AesKeyStrength
+import net.lingala.zip4j.model.enums.EncryptionMethod
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+
+
+interface WalletBackup {
+    fun run(key: CharArray): File
+
+    companion object {
+        fun defaultStrategy(cd: File): WalletBackup = CompressionOnlyBackup.defaultStrategy(cd)
+    }
+}
+
+val BACKUP_FILES = arrayOf("tari_wallet_db.sqlite3")
+
+private class PasswordProtectedZipBackup(private val cd: File) : WalletBackup {
```

Comment:
Let's make sure we have full understanding of how the library works in terms of the way it encrypts the file so that  we can have exactly the same on iOS.

Reply:
Please see an earlier comment on abbreviations/acronyms for param name `cd`.

Reply:
all the acronyms concerning this will be changed, yes

no need to mark every entry, one mark is sufficient :)

Reply:
ok:)